index.js

```diff
@@ -137,28 +137,37 @@ HtmlWebpackPlugin.prototype.apply = function (compiler) {
       // Allow plugins to change the html before assets are injected
       .then(function (html) {
         var pluginArgs = {html: html, assets: assets, plugin: self, outputName: self.childCompilationOutputName};
-        return applyPluginsAsyncWaterfall('html-webpack-plugin-before-html-processing', pluginArgs)
-          .then(function () {
-            return pluginArgs.html;
-          });
+        return applyPluginsAsyncWaterfall('html-webpack-plugin-before-html-processing', pluginArgs);
       })
-      .then(function (html) {
+      .then(function (result) {
+        var html = result.html;
+        var assets = result.assets;
+        var chunks = result.chunks;
         // Prepare script and link tags
         var assetTags = self.generateAssetTags(assets);
-        var pluginArgs = {head: assetTags.head, body: assetTags.body, plugin: self, chunks: chunks, outputName: self.childCompilationOutputName};
+        var pluginArgs = {head: assetTags.head, body: assetTags.body, plugin: self, chunks: chunks, outputName: self.childCompilationOutputName, assets: assets, html: html};
```

Comment:
Why did you add html here?
We don't want people to modify assets or html at this points anymore - this would cause a mess.